python/graphstorm/model/gnn_encoder_base.py

```diff
@@ -102,6 +102,25 @@ def dist_inference(self, g, get_input_embeds, batch_size, fanout,
         return dist_inference(g, self, get_input_embeds, batch_size, fanout,
                             edge_mask=edge_mask, task_tracker=task_tracker)
 
+def fix_for_wholegraph(g, input_nodes, not_end_of_loader):
```

Comment:
not_end_of_loader is not necessary.
The input_nodes can be an empty dict.

Reply:
Can you add some unit test for fix_for_wholegraph?

Reply:
Done.